atlp-rwanda/atlp-pulse-mobile
# Real time notifications

## What does this PR do?
This pull request (PR) implements the functionality to fetch real-time trainee notifications using GraphQL Subscription. This allows the notifications for trainees to be automatically updated when a new notification is encountered, without requiring the user to manually reload the application.

## Description of Task to be completed

- Real-Time Notifications: Added a NotificationSubscription and PUSH_NOTIFICATION_SUB subscription for real-time notification updates. This means that when a new notification is sent to a trainee, it is immediately pushed to their user interface without the need to reload the application.

- Unread Notification Count: Created a Notification Provider that will display the number of unread notifications on the app's top bar. This provides users with a quick, at-a-glance view of any outstanding notifications.

## How should this be manually tested?

1. Real-time Notifications:

Log in as a trainee and navigate to the notifications page.
Verify that any new notifications are displayed directly, without the need to reload the page.


2. Notification Counter:

Verify that the number of unread notifications is displayed above the notification icon in the app's top bar.

## Screen shots

<img src="https://github.com/user-attachments/assets/0bfb6b19-f671-4051-92d6-f582b1e3a8cb" alt="Description" width="350" />
<img src="https://github.com/user-attachments/assets/e72131e0-2b75-454a-8d77-85f95feb2601" alt="Description" width="350" />
